qt/scientific_interfaces/Indirect/InelasticDataManipulationElwinTab.cpp

```diff
@@ -201,19 +202,25 @@ void InelasticDataManipulationElwinTab::runWorkspaceInput() {
   std::string inputGroupWsName = "IDA_Elwin_Input";
 
   // Load input files
-  std::string inputWorkspacesString = m_view->getCurrentPreview().toStdString();
+  std::string inputWorkspacesString;
+  for (WorkspaceID i = 0; i < m_dataModel->getNumberOfWorkspaces(); ++i) {
 
+    auto workspace = m_dataModel->getWorkspace(i);
+    auto spectra = m_dataModel->getSpectra(i);
+    auto spectraWS = m_model->createGroupedWorkspaces(workspace, spectra);
+    inputWorkspacesString += spectraWS + ",";
+  }
   // Group input workspaces
   m_model->setupGroupAlgorithm(m_batchAlgoRunner, inputWorkspacesString, inputGroupWsName);
 
-  m_model->setupElasticWindowMultiple(m_batchAlgoRunner, m_view->getCurrentPreview(), inputGroupWsName,
+  m_model->setupElasticWindowMultiple(m_batchAlgoRunner, "ELWIN_workspace_output", inputGroupWsName,
                                       m_view->getLogName(), m_view->getLogValue());
 
   connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(unGroupInput(bool)));
   m_batchAlgoRunner->executeBatchAsync();
 
   // Set the result workspace for Python script export
-  m_pythonExportWsName = (m_view->getCurrentPreview() + "_elwin_eq2").toStdString();
+  m_pythonExportWsName = "ELWIN_workspace_output_elwin_eq2";
```

Comment:
Can I just double check that it's correct that this is hardcoded with a fixed string?

Reply:
Yes when outputing via workspace input that output is hardcoded

Reply:
@Harrietbrown @martyngigg Shouldn't have the instrument & runnumber etc?

Reply:
There can be potentially multiple run numbers or instruments as they can be selected together in the input

Reply:
What was `m_view->getCurrentPreview()` returning before?

Reply:
it was the workspace that was in the preview plot

Reply:
So the name was misleading as it could contain inputs from multiple workspaces yet the name suggested a single source? Is this name only used if you generate a script from Python so in that case it's easy for someone to change to something more meaningful?

Reply:
What do you mean if they generate it from python? do you mean if they get the workspace history of the output?

Reply:
I was just thinking of the name of the variable and whether it was linked to something that is output to a script? Isn't there a button on the interface to generate one?

Reply:
There isn;t a button to generate a script in elwin

Reply:
Ah okay. Could we rename this variable to avoid the confusion?

Reply:
Which variable? do you mean set the name for the suffix before using it for the name?

Reply:
I was more meaning `m_pythonExportWsName` feels like it is suggesting it is used in Python somehow but if it's not can we make it clearer what it is used for?

Reply:
is used in all of the indirectTabs where a python export is used. it would not be simple to replace that for all of them. currently it also works for all the other outputs whenever it is needed.

Reply:
Okay. I'll defer to @AnthonyLim23 & @MohamedAlmaki for the future to decide where to take this so we can hand this over before you go. Thanks.